### PR TITLE
Use architecture pointer size for AddConstructor

### PR DESCRIPTION
**Pull request by l4l**

### CODE REVIEW

File: src/backend.rs [hunk 1]
```diff
@@ -132,10 +132,16 @@ impl AddConstructor for ObjectProduct {
         let init_array_section =
             self.object
                 .add_section(segment.to_vec(), b".init_array".to_vec(), SectionKind::Data);
+        let addres_size = self
```
Comment (bjorn3) [inline src/backend.rs:135]:
```suggestion
        let address_size = self
```
nit: typo

**Review by bjorn3 — APPROVED** [review]
Thanks!

File: src/backend.rs [hunk 2]
```diff
@@ -132,10 +132,16 @@ impl AddConstructor for ObjectProduct {
...
+            .object
+            .architecture()
+            .address_size()
+            .expect("address_size must be known")
+            .bytes();
         self.object.append_section_data(
             init_array_section,
             &std::iter::repeat(0)
-                .take(8 /*FIXME pointer size*/)
+                .take(addres_size.into())
```
Comment (bjorn3) [inline src/backend.rs:144]:
```suggestion
                .take(address_size.into())
```

File: src/backend.rs [hunk 3]
```diff
@@ -144,7 +150,7 @@ impl AddConstructor for ObjectProduct {
                 init_array_section,
                 object::write::Relocation {
                     offset: 0,
-                    size: 64, // FIXME pointer size
+                    size: addres_size * 8,
```
Comment (bjorn3) [inline src/backend.rs:153]:
```suggestion
                    size: address_size * 8,
```

File: src/backend.rs (hunk 2 above)
Comment (l4l) [reply src/backend.rs]: Oops, thanks

**Pull request merged**